src/OpenLoco/src/World/Station.cpp

```diff
@@ -873,6 +880,186 @@ namespace OpenLoco
+
+        for (auto i = 0U; i < station->stationTileSize; ++i)
+        {
+            auto& pos = station->stationTiles[i];
+            StationElement* elStation = getStationElement(pos);
+            if (elStation == nullptr || elStation->isGhost())
+            {
+                continue;
+            }
+            switch (elStation->stationType())
+            {
+                case StationType::trainStation:
+                {
+                    auto* elTrack = elStation->prev()->as<TrackElement>();
+                    if (elTrack == nullptr)
+                    {
+                        break;
+                    }
+                    auto* trackObj = ObjectManager::get<TrackObject>(elTrack->trackObjectId());
+                    station->flags |= trackObj->hasFlags(TrackObjectFlags::unk_02)
+                        ? StationFlags::transportModeRoad
+                        : StationFlags::transportModeRail;
+                    acceptedCargoTypes = 0xFFFFFFFFU;
+                    break;
+                }
+                case StationType::roadStation:
+                {
+                    // Loop from start as road element can be variable number of elements below station
+                    auto tile = TileManager::get(pos);
+                    for (auto& el2 : tile)
+                    {
+                        if (&el2 == reinterpret_cast<TileElement*>(&elStation))
+                        {
+                            break;
+                        }
+                        auto* elRoad = el2.as<RoadElement>();
+                        if (elRoad == nullptr)
+                        {
+                            continue;
+                        }
+                        if (elRoad->baseZ() != elStation->baseZ())
+                        {
+                            continue;
+                        }
+                        if (elRoad->isAiAllocated() || elRoad->isGhost())
+                        {
+                            continue;
+                        }
+                        auto* roadObj = ObjectManager::get<RoadObject>(elRoad->roadObjectId());
+                        station->flags |= roadObj->hasFlags(RoadObjectFlags::unk_01)
+                            ? StationFlags::transportModeRail
+                            : StationFlags::transportModeRoad;
+                    }
+                    auto* roadStationObj = ObjectManager::get<RoadStationObject>(elStation->objectId());
+                    if (roadStationObj->hasFlags(RoadStationFlags::passenger))
+                    {
+                        acceptedCargoTypes |= 1U << roadStationObj->cargoType;
+                    }
+                    else if (roadStationObj->hasFlags(RoadStationFlags::freight))
+                    {
+                        // Exclude passengers from this station type
+                        acceptedCargoTypes |= 0xFFFFFFFFU & ~(1U << roadStationObj->cargoType);
```

Comment:
Not sure how this is supposed to work. Won't it just enable all cargo types _except_ the relevant one?

Reply:
That is exactly what it will do. This bit of code is identical to another bit in the cargo calc code and it basically makes it clear that passengers can't load at freight stations

Reply:
Been mulling this over. I think what confuses me is the fact that `|=` is used, rather than a regular assignment. By using `0xFFFFFFFFU`, any previously unset bits will be reset, anyway.

Do I understand correctly that, if `roadStationObj->hasFlags(RoadStationFlags::freight)`, `~(1U << roadStationObj->cargoType)` will mean all non-freight cargo, i.e. passengers? It's a little roundabout, but it makes sense to not have to look up the passenger index, I suppose.

I suppose the `0xFFFFFFFFU` mask is needed because the word length of `~(1U << roadStationObj->cargoType)` isn't necessarily 32 bits?

Reply:
I've made it match the style of the other verison of the loop and added a comment as to the differences between the two loops. Hope that makes things clear enough.